src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/PPAnalyticsSubmitter.swift

```diff
@@ -36,120 +37,145 @@ final class PPAnalyticsSubmitter: PPAnalyticsSubmitting {
-				
-				guard let self = self else {
-					Log.warning("Analytics submission abord due fail at creating strong self", log: .ppa)
-					completion?(.failure(.generalError))
-					return
-				}
-				
-				// Check configuration parameter
-				if Double.random(in: 0...1) > configuration.privacyPreservingAnalyticsParameters.common.probabilityToSubmit {
-					Log.warning("Analytics submission abord due to randomness", log: .ppa)
-					completion?(.failure(.probibilityError))
-					return
-				}
-				
-				// Last submission check
-				if self.submissionWithinLast23Hours {
-					Log.warning("Analytics submission abord due to submission last 23 hours", log: .ppa)
-					completion?(.failure(.submission23hoursError))
-					return
-				}
-				
-				// Onboarding check
-				if self.onboardingCompletedWithinLast24Hours {
-					Log.warning("Analytics submission abord due to onboarding completed last 24 hours", log: .ppa)
-					completion?(.failure(.onboardingError))
-					return
-				}
-				
-				// App Reset check
-				if self.appResetWithinLast24Hours {
-					Log.warning("Analytics submission abord due to app resetted last 24 hours", log: .ppa)
-					completion?(.failure(.appResetError))
-					return
-				}
-				
-				self.hoursSinceTestResultToSubmitKeySubmissionMetadata = configuration.privacyPreservingAnalyticsParameters.common.hoursSinceTestResultToSubmitKeySubmissionMetadata
-				self.hoursSinceTestRegistrationToSubmitTestResultMetadata = configuration.privacyPreservingAnalyticsParameters.common.hoursSinceTestRegistrationToSubmitTestResultMetadata
-				self.probabilityToSubmitExposureWindows = configuration.privacyPreservingAnalyticsParameters.common.probabilityToSubmitExposureWindows
-
-				if let token = ppacToken {
-					// Submit analytics data with injected ppac token
-					self.submitData(with: token, completion: completion)
-				} else {
-					self.generatePPACAndSubmitData(completion: completion)
-				}
+		// Sink on the app configuration if something has changed. But do this in background and wait for the result before continue.
+		dispatchGroupSubmission.enter()
+		self.configurationProvider.appConfiguration().receive(on: DispatchQueue.global(qos: .background).ocombine).sink { [ weak self] configuration in
+			let ppaConfigData = configuration.privacyPreservingAnalyticsParameters.common
+			self?.probabilityToSubmitPPAUsageData = ppaConfigData.probabilityToSubmit
+			self?.hoursSinceTestResultToSubmitKeySubmissionMetadata = ppaConfigData.hoursSinceTestResultToSubmitKeySubmissionMetadata
+			self?.hoursSinceTestRegistrationToSubmitTestResultMetadata = ppaConfigData.hoursSinceTestRegistrationToSubmitTestResultMetadata
+			self?.probabilityToSubmitExposureWindows = ppaConfigData.probabilityToSubmitExposureWindows
+			self?.dispatchGroupSubmission.leave()
+		}.store(in: &subscriptions)
+		
+		// If we have the app config, we continue...
+		dispatchGroupSubmission.notify(queue: .global(qos: .background)) { [weak self] in
```

Comment:
Why is the dispatch group needed? Can't the logic not use be called in the completion of the appconfig sink?

Reply:
Honestly I though it would give the code a better structure to not do everything in the completion of the appconfig sink.

Reply:
i removed the dispatchgroup